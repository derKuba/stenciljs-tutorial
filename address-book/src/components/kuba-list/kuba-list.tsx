import { Component, Host, h } from "@stencil/core";

import addressStore from "../../store/address-store";

@Component({
  tag: "kuba-list",
  styleUrl: "kuba-list.css",
  shadow: true,
})
export class KubaList {
  deleteContact = (id: string) => {
    const filteredArray = addressStore.contacts.filter(
      (item) => item.id !== id,
    );
    addressStore.contacts = filteredArray;
  };

  render() {
    return (
      <Host>
        <stencil-route-link id="addressbook_new_contact" url="/contact" activeClass="link-active">
          Neuer Kontakt
        </stencil-route-link>
        <hr />

        <kuba-table-options
          delete={this.deleteContact}
          key={addressStore.contacts.length}
        >
          <kuba-table-options-head>
            <option value="id" />
            <option value="name" />
            <option value="vorname" />
            <option value="aktion" />
          </kuba-table-options-head>
          <kuba-table-options-body>
            {addressStore.contacts.map((row) => [
              <option value={row.id} />,
              <option value={row.firstName} />,
              <option value={row.lastName} />,
              <option value={row.id} />,
            ])}
          </kuba-table-options-body>
        </kuba-table-options>
        <hr />
      </Host>
    );
  }
}
