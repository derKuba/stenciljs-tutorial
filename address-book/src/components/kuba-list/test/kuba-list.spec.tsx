import { newSpecPage } from "@stencil/core/testing";
import { KubaList } from "../kuba-list";

describe("kuba-list", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [KubaList],
      html: `<kuba-list></kuba-list>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
