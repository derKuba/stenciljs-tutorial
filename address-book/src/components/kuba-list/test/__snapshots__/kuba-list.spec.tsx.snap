// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kuba-list renders 1`] = `
<kuba-list>
  <mock:shadow-root>
    <stencil-route-link activeclass="link-active" url="/contact">
      Neuer Kontakt
    </stencil-route-link>
    <hr>
    <kuba-table-options>
      <kuba-table-options-head>
        <option value="id"></option>
        <option value="name"></option>
        <option value="vorname"></option>
        <option value="aktion"></option>
      </kuba-table-options-head>
      <kuba-table-options-body></kuba-table-options-body>
    </kuba-table-options>
    <hr>
  </mock:shadow-root>
</kuba-list>
`;
